### PR TITLE
Section: Why was the database redesign done

### PR DESCRIPTION
**Pull request by Pattrigue**
Besides a general review, we would like to know if there's anything else we should add to the section.

### CODE REVIEW

File: sections/Chapter 5 Sprint 3/db_design_redone.tex [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+As previously mentioned in section \ref{currentState}, the code from last year was left in an unfinished and undocumented state. We estimated that it would be faster and easier to scrap it and start over rather than spending time trying to understand and document it.
+However, the database design seemed good enough to work with despite its flaws. 
+We therefore decided to keep it as we were under time constraints.
+
+In order to communicate with the database from the C\# application, we opted to use EF Core.
+Because EF Core is designed with the Code First approach in mind, we are unable to simply make SQL queries in the code itself as the old group did. 
+Instead, we had to model the current database design using C\# classes. 
+
+This caused problems as the current design did not translate well into an object-oriented approach.
+To execute what would otherwise be considered a relative simple SQL query, we had to write a ton of code which ultimately defeated the purpose of the Code First approach.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 5 Sprint 3/db_design_redone.tex:12]:
```suggestion
To execute what would otherwise be considered a relatively simple SQL query, we had to write additional code which ultimately defeated the purpose of the Code First approach. 
```

File: sections/Chapter 5 Sprint 3/db_design_redone.tex [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+However, the database design seemed good enough to work with despite its flaws. 
+We therefore decided to keep it as we were under time constraints.
+
+In order to communicate with the database from the C\# application, we opted to use EF Core.
+Because EF Core is designed with the Code First approach in mind, we are unable to simply make SQL queries in the code itself as the old group did. 
+Instead, we had to model the current database design using C\# classes. 
+
+This caused problems as the current design did not translate well into an object-oriented approach.
+To execute what would otherwise be considered a relative simple SQL query, we had to write a ton of code which ultimately defeated the purpose of the Code First approach. 
+The most important method for the WordCount controller, the POST method, can be seen in code snippet \ref{lst:old_controller} for the old database design.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 5 Sprint 3/db_design_redone.tex:13]: Why is it the most important? Is fetching data from the database not equally important?

File: sections/Chapter 5 Sprint 3/db_design_redone.tex [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    foreach (Article article in enumerable)
+    {
+        Console.WriteLine($"Added {article.Title}");
+    }
+
+    return Ok(message.ToString());
+}
+\end{lstlisting}
+    
+By comparing the semantics and overall size of code snippet \ref{lst:old_controller} and \ref{lst:new_controller}, it should be apparent that the redesigned database is much easier to work with using EF Core, as it removed a lot of unnecessary complexity in the code.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 5 Sprint 3/db_design_redone.tex:142]: How do we compare semantics?
```suggestion
By comparing the overall size of code snippet \ref{lst:old_controller} and \ref{lst:new_controller}, it should be apparent that the redesigned database is much easier to work with using EF Core, as it removed a lot of unnecessary complexity in the code.
```

File: sections/Chapter 5 Sprint 3/db_design_redone.tex [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+However, despite the many advantages from the redesign, it has a major memory issue.
+In our rush to quickly implement the redesign, we unfortunately overlooked how to effectively store words from articles.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 5 Sprint 3/db_design_redone.tex:144]:
```suggestion
In our rush to quickly implement the redesign, we, unfortunately, overlooked how to effectively store words from articles. 
```

File: sections/Chapter 5 Sprint 3/db_design_redone.tex [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    return Ok(message.ToString());
+}
+\end{lstlisting}
+    
+By comparing the semantics and overall size of code snippet \ref{lst:old_controller} and \ref{lst:new_controller}, it should be apparent that the redesigned database is much easier to work with using EF Core, as it removed a lot of unnecessary complexity in the code.
+However, despite the many advantages from the redesign, it has a major memory issue.
+In our rush to quickly implement the redesign, we unfortunately overlooked how to effectively store words from articles. 
+
+Currently, we just store every single word in the database without regard for duplicates, which takes up way more space than necessary.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 5 Sprint 3/db_design_redone.tex:146]:
```suggestion
Currently, we just store every single word in the database without regard for duplicates, which takes up way more space than necessary. Thus it is not of the third normal form, and thus not in BCNF either.
```

**Review by RasmusHoeyerHansen — CHANGES_REQUESTED** [review]
Very good! Have a few suggestions and comments. It would be really good to give a concrete evaluation of our database design using methods from DBS course.  However, with minor corrections, the current text describes the approach well.

File: sections/Chapter 5 Sprint 3/db_design_redone.tex (hunk 2 above)
Comment (Pattrigue) [reply sections/Chapter 5 Sprint 3/db_design_redone.tex:13]: Maybe rephrase it to "the most complex" or "most comprehensive"?

File: sections/Chapter 5 Sprint 3/db_design_redone.tex (hunk 3 above)
Comment (Pattrigue) [reply sections/Chapter 5 Sprint 3/db_design_redone.tex:142]: I mostly meant what functions actually do with the old one clearly doing way more work than the new one.
If possible it would be nice to still have that included somehow.

File: sections/Chapter 5 Sprint 3/db_design_redone.tex (hunk 4 above)
Comment (Pattrigue) [reply sections/Chapter 5 Sprint 3/db_design_redone.tex:144]: Not an interposed sentence so no commas.

File: sections/Chapter 5 Sprint 3/db_design_redone.tex (hunk 5 above)
Comment (Pattrigue) [reply sections/Chapter 5 Sprint 3/db_design_redone.tex:146]: This is a good idea but I don't like using the word "thus" twice in a row.

File: sections/Chapter 5 Sprint 3/db_design_redone.tex [hunk 6]
```diff
@@ -0,0 +1,148 @@
+\section{Database redesign}
+
+As previously mentioned in section \ref{currentState}, the code from last year was left in an unfinished and undocumented state. We estimated that it would be faster and easier to scrap it and start over rather than spending time trying to understand and document it.
```
Comment (RasmusHoeyerHansen) [inline sections/Chapter 5 Sprint 3/db_design_redone.tex:3]:
```suggestion
As previously mentioned in section \ref{currentState}, the code from last year was left in an unfinished and undocumented state. We estimated that starting over would give more value to the pipeline, rather than spending time trying to understand and document it.
```

File: sections/Chapter 5 Sprint 3/db_design_redone.tex (hunk 2 above)
Comment (RasmusHoeyerHansen) [reply sections/Chapter 5 Sprint 3/db_design_redone.tex:13]:
```suggestion
The most complex method for the WordCount controller, the POST method, can be seen in code snippet \ref{lst:old_controller} for the old database design.
```

**Review by RasmusHoeyerHansen — CHANGES_REQUESTED** [review]
Looks good, my suggestions can be reevaluated.

**Review by RasmusHoeyerHansen — APPROVED** [review]
